src/main/java/ch/njol/skript/expressions/ExprIP.java

```diff
@@ -24,97 +24,47 @@
 
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
-import org.bukkit.event.player.PlayerLoginEvent;
 import org.eclipse.jdt.annotation.Nullable;
 
-import ch.njol.skript.ScriptLoader;
 import ch.njol.skript.Skript;
 import ch.njol.skript.doc.Description;
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
-import ch.njol.skript.lang.Expression;
-import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.util.SimpleExpression;
-import ch.njol.util.Kleenean;
 
 @Name("IP")
-@Description("The IP address of a player.")
+@Description("The IP address of player(s).")
 @Examples({"IP-ban the player # is equal to the next line",
 		"ban the IP-address of the player",
 		"broadcast \"Banned the IP %IP of player%\""})
-@Since("1.4, 2.2-dev26 (when used in connect event)")
-public class ExprIP extends SimpleExpression<String> {
+@Since("1.4, 2.2-dev35")
+public class ExprIP extends SimplePropertyExpression<Player, String> {
 	
 	static {
-		Skript.registerExpression(ExprIP.class, String.class, ExpressionType.PROPERTY, "IP[s][( |-)address[es]] of %players%",
-				"%players%'[s] IP[s][( |-)address[es]]");
+		register(ExprIP.class, String.class, "IP[s][( |-)address[es]]", "players");
 	}
-	
-	@SuppressWarnings("null")
-	private Expression<Player> players;
-	private boolean connectEvent;
-	
-	@SuppressWarnings({"null", "unchecked"})
-	@Override
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		players = (Expression<Player>) exprs[0];
-		connectEvent = ScriptLoader.isCurrentEvent(PlayerLoginEvent.class);
-		return true;
-	}
-	
-	@Override
-	@Nullable
-	protected String[] get(Event e) {

```

Comment:
While convering to SimplePropertyExpression is ok, breaking IP retrieval when this is used for event-player in connect event. At least with some of supported versions, Spigot does not provide IP for such an player yet.